### PR TITLE
When showing or finding offers, ignore those with errors and include store in output

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

Some usability tweak to CMR show/find output based on initial user feedback.
1. If an offer is bad (eg application deleted, skip it instead of aborting find)
2. show store in show/find output

## QA steps

bootstrap and create offers
find show/find to see new output
delete offered application and find should still work

## Bug reference

https://bugs.launchpad.net/juju/+bug/1702901
https://bugs.launchpad.net/juju/+bug/1702899

### CODE REVIEW

File: cmd/juju/crossmodel/showformatter.go [hunk 1]
```diff
@@ -37,10 +38,19 @@ func formatOfferedEndpointsTabular(writer io.Writer, all map[string]ShowOfferedA
...
-	for name, one := range all {
-		offerName := name
+	for urlStr, one := range all {
+		url, err := crossmodel.ParseApplicationURL(urlStr)
+		if err != nil {
+			return err
+		}
+		store := url.Source
+		if store == "" {
+			store = "local"
```
Comment (axw) [inline cmd/juju/crossmodel/showformatter.go]: what does "local" mean in this case? local to the controller that we're querying? what other values might it take on?

Comment (wallyworld) [inline cmd/juju/crossmodel/showformatter.go]: It's just a label which probably could be improved. Store should never be empty (anymore) so I should just remove this bit.

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**